File: solutions/javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(max) {
    this._n = max;
  }

  get sumOfSquares() {
    return (this._n + 3 * this._n ** 2 + 2 * this._n ** 3) / 6;
  }

  get squareOfSum() {
    return ((this._n + this._n ** 2) / 2) ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: solutions/ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(max)
    @n = max
  end

  def square_of_sum
    ((@n + @n ** 2) / 2) ** 2
  end

  def sum_of_squares
    (@n + 3 * @n ** 2 + 2 * @n ** 3) / 6
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
